libd/src/d/parser/statement.d

```diff
@@ -406,7 +406,33 @@ AstStatement parseStatement(ref TokenRange trange) {
 			return new AstSynchronizedStatement(location, statement);
 		
 		case Mixin :
-			return trange.parseMixin!AstStatement();
+			//could be an expression so if we cannot parse it as a Statement fall back to expression
+			trange.popFront();
+			trange.match(TokenType.OpenParen);
+
+			auto expr = trange.parseAssignExpression();
+			auto mixinEndLoc = trange.front.location;
+
+			trange.match(TokenType.CloseParen);
+
+			if (trange.front.type == TokenType.Semicolon) {
+				location.spanTo(trange.front.location);
+				trange.popFront();
+
+				return new d.ast.conditional.Mixin!AstStatement(location, expr);
```

Comment:
the fully qualified name is needed because if would conflict with the TokenType.Mixin